src/industry_cmd.cpp

```diff
@@ -2246,12 +2247,21 @@ static void UpdateIndustryStatistics(Industry *i)
 				i->last_prod_year = _cur_year;
 				pct = min(i->this_month_transported[j] * 256 / i->this_month_production[j], 255);
 			}
-			i->last_month_pct_transported[j] = pct;
+			for (size_t k = 23; k >= 1; k--){
```

Comment:
Please use lengthof() or a declared constant instead of magic number 23.

Reply:
Fixed.